### PR TITLE
RandomForestClassifier for python

### PR DESCRIPTION
**Pull request by krinart**

### CODE REVIEW

File: m2cgen/ast.py [hunk 1]
```diff
@@ -48,18 +48,53 @@ def __str__(self):
...
 
 
-class VectorExpr(NumExpr):
+# Vector Expressions.
+
+class VectorExpr(Expr):
     is_vector_output = True
 
+
+class VectorVal(VectorExpr):
```
Comment (izeigerman) [inline m2cgen/ast.py]: Can we call it `VectorValExpr` to maintain consistency with other AST members.

File: m2cgen/interpreters/interpreter.py [hunk 2]
```diff
@@ -5,6 +5,9 @@
 
 class BaseInterpreter:
 
+    with_vectors = False
+    with_linear_algebra = False
```
Comment (izeigerman) [inline m2cgen/interpreters/interpreter.py]: Aren't these 2 attributes interchangeable? I can't find a use case when we may have one without another.

File: m2cgen/interpreters/python/interpreter.py [hunk 3]
```diff
@@ -17,6 +19,23 @@ def interpret(self, expr):
             last_result = self._do_interpret(expr)
             self._cg.add_return_statement(last_result)
 
+        if self.with_linear_algebra or self.with_vectors:
+            self._cg.prepend_code_line("import numpy as np")
```
Comment (izeigerman) [inline m2cgen/interpreters/python/interpreter.py]: Code leak into interpreter? I thought we should delegate all code related logic to Code Generator instead.

File: m2cgen/interpreters/interpreter.py (hunk 2 above)
Comment (krinart) [reply m2cgen/interpreters/interpreter.py]: Done

File: m2cgen/ast.py (hunk 1 above)
Comment (krinart) [reply m2cgen/ast.py]: Ack

**Comment by izeigerman** [comment]
🎉 🥇 👍

**Pull request merged**